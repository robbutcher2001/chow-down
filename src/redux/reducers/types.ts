export interface Message {
    user: string
    message: string
}

export interface HelloState {
    hello_messages: Message[]
}

export const SAY_HELLO = 'SAY_HELLO'
export const DELETE_HELLO = 'DELETE_HELLO';

interface SayHelloAction {
    type: typeof SAY_HELLO
    rob_payload: Message
}

interface DeleteHelloAction {
    type: typeof DELETE_HELLO
}

export type SayHelloTypes = SayHelloAction | DeleteHelloAction