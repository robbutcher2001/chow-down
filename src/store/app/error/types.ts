import { Action } from 'redux';

export enum ErrorActionTypes {
  UNEXPECTED_SERVER_ERROR = '@@common/UNEXPECTED_SERVER_ERROR',
  UNEXPECTED_RESPONSE = '@@common/UNEXPECTED_RESPONSE',
  CLEAR_ERROR = '@@common/CLEAR_ERROR'
}

export interface ErrorState {
  readonly message: string
}

export interface ErrorMessageApiResponse {
  message: string
}

export interface ErrorApiResponse extends Action {
  type: ErrorActionTypes,
  json: ErrorMessageApiResponse
}