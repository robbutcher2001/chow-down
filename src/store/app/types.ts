import { Action } from 'redux';

export enum AppActionTypes {
    UNEXPECTED_SERVER_ERROR = '@@common/UNEXPECTED_SERVER_ERROR',
    UNEXPECTED_RESPONSE = '@@common/UNEXPECTED_RESPONSE',
    CLEAR_ERROR = '@@common/CLEAR_ERROR'
}

export interface AppState {
    readonly error: string
}

export interface ErrorMessageApiResponse {
    message: string
}

export interface ErrorApiResponse extends Action {
    type: AppActionTypes,
    json: ErrorMessageApiResponse
}