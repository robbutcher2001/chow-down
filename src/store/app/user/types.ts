import { Action } from 'redux';

export enum UserActionTypes {
  SET_USER_IS_SELECTING_DAY = '@@common/SET_USER_IS_SELECTING_DAY',
  CLEAR_USER_IS_SELECTING_DAY = '@@common/CLEAR_USER_IS_SELECTING_DAY'
}

export interface UserState {
  readonly selectedDay: string
}

export interface UserAction extends Action {
  type: UserActionTypes,
  day: string
}