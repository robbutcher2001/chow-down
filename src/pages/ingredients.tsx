import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import { GlobalState } from '../store';
import { Ingredient, GetIngredientsApiRequest, PostIngredientApiRequest } from '../store/domain/ingredients/types';
import { getIngredientsRequest, postIngredientsRequest } from '../store/domain/ingredients/actions';

import Form from '../components/form';
import InputBox from '../components/input-box';

interface StateProps {
    error: string,
    failure: string,
    ingredients: Ingredient[],
    ui: {
        pending: {
            get: boolean,
            post: boolean
        }
    }
};

interface DispatchProps {
    getIngredients: () => GetIngredientsApiRequest,
    postIngredient: (form: object) => PostIngredientApiRequest
};

interface OwnProps { };

interface OwnState { };

type CombinedProps = StateProps & DispatchProps & OwnProps;

class IngredientsPage extends Component<CombinedProps, OwnState> {
    constructor(props: CombinedProps) {
        super(props);
    }

    requestIngredients = () => this.props.getIngredients();
    addIngredient = (form: object) => this.props.postIngredient(form);

    componentDidMount = () => this.props.getIngredients();

    render = () => {
        console.log(this.props.ui.pending.post);
        return (
            <div>
                <h4>List ingredients</h4>
                <button onClick={this.requestIngredients}>
                    Refresh ingredients
                </button>
                {this.props.ui.pending.get &&
                    <div style={{ color: 'red' }}>Getting..</div>
                }
                <ul>
                    {this.props.ingredients.map((ingredient, index) =>
                        <li key={index}>{ingredient.name}</li>
                    )}
                </ul>
                <h4>Add ingredient</h4>
                <Form
                    dispatch={this.addIngredient}
                    submitText='Add ingredient'>
                    <InputBox
                        name='name'
                        placeholderText='New ingredient name'
                    />
                </Form>
                <div>{this.props.error}</div>
                {this.props.ui.pending.post &&
                    <div>Adding your new ingredient..</div>
                }
                {this.props.failure &&
                    <div style={{color: 'red'}}>{this.props.failure}</div>
                }
            </div>
        );
    }
};

const mapStateToProps = ({ app, domain, ui }: GlobalState, ownProps: OwnProps): StateProps => ({
    //TODO: move application-wide errors to footer component for toast notification (feed down through page container)
    error: app.error,
    failure: domain.ingredient.failure,
    ingredients: domain.ingredient.ingredients,
    ui: {
        pending: {
            get: ui.ingredient.getPending,
            post: ui.ingredient.postPending
        }
    }
});

const mapDispatchToProps = (dispatch: Dispatch, ownProps: OwnProps): DispatchProps => ({
    getIngredients: () => dispatch(getIngredientsRequest()),
    postIngredient: (form: object) => dispatch(postIngredientsRequest(form))
});

export default connect<StateProps, DispatchProps, OwnProps, GlobalState>
    (mapStateToProps, mapDispatchToProps)(IngredientsPage);