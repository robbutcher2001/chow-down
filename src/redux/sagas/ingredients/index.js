'use strict';

import { all, takeLatest } from 'redux-saga/effects';

import getSaga from './get';
import postSaga from './post';

export default function* watcherSaga() {
    console.log('[saga] Started ingredients saga');
    yield all([
        takeLatest('GET_INGREDIENTS_REQUEST', getSaga),
        takeLatest('POST_INGREDIENT_REQUEST', postSaga)
    ]);
}