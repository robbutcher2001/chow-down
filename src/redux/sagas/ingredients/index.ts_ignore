'use strict';

import { all, takeLatest } from 'redux-saga/effects';

import { Actions } from '../../../globals/constants';

import getSaga from './get';
import postSaga from './post';

export default function* watcherSaga() {
    console.log('[saga] Started ingredients saga');
    yield all([
        takeLatest(Actions.ingredients.GET_INGREDIENTS_REQUEST, getSaga),
        takeLatest(Actions.ingredients.POST_INGREDIENT_REQUEST, postSaga)
    ]);
}