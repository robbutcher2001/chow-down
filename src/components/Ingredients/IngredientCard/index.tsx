import React from 'react';

import styled from 'styled-components';

import { Ingredient } from '../../../store/domain/ingredients/types';

interface IngredientCardProps {
  ingredient: Ingredient
};

const IngredientCard = styled.li`
  background: rgba(171, 184, 195, 0.25);
  border-left-style: solid;
  border-left-width: 0.35rem;
  border-left-color: #0693E3;
  margin: 0.5rem;
  padding: 0.75rem;
  border-radius: 4px;
  box-shadow: 0px 2px 2px 0 rgba(0,0,0,0.2);
  box-sizing: border-box;
`

export default (props: IngredientCardProps) => (
  <IngredientCard>
    {props.ingredient.ingredient}
  </IngredientCard>
);