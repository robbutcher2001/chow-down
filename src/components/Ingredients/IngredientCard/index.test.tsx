import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../../../themes';
import IngredientCard from '.';
import { Ingredient } from '../../../store/domain/ingredients/types';

const mockIngredient: Ingredient = {
  id: '123',
  name: 'fake_ingredient'
};

test('IngredientCard basic snapshot render', () => {
  const ingredientCard = renderer.create(
    <ThemeProvider theme={theme}>
      <IngredientCard ingredient={mockIngredient} />
    </ThemeProvider>
  );

  expect(ingredientCard.toJSON()).toMatchSnapshot();
});

test('IngredientCard correct tag content assertion', () => {
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <IngredientCard ingredient={mockIngredient} />
    </ThemeProvider>
  );

  expect(getByText(/fake_ingredient/).textContent).toEqual('fake_ingredient');
});