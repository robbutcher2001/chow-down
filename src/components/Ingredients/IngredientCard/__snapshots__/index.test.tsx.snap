// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientCard basic snapshot render 1`] = `
<li
  className="sc-bdVaJa eIQMUN"
>
  fake_ingredient
</li>
`;
